Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -191,24 +243,62 @@ internal class ResponseAccumulator: HTTPClientResponseDelegate {
     }
 }
 
-/// This delegate is strongly held by the HTTPTaskHandler
-/// for the duration of the HTTPRequest processing and will be
-/// released together with the HTTPTaskHandler when channel is closed
+/// HTTPClientResponseDelegate allows to receive notification about request processing and to control how reponse parts are processed.
+///
+///  - note: This delegate is strongly held by the `HTTPTaskHandler`
+///          for the duration of the `Request` processing and will be
+///          released together with the `HTTPTaskHandler` when channel is closed.
 public protocol HTTPClientResponseDelegate: AnyObject {
     associatedtype Response
 
+    /// Called when request `Head` is sent. Will be called once.
```

Comment:
Called when the request head is sent. Will be called once.

Reply:
fixed, thanks!